presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -502,6 +503,9 @@ public IndexMetadata getIndexMetadata(String index)
             catch (IOException e) {
                 throw new PrestoException(ELASTICSEARCH_INVALID_RESPONSE, e);
             }
+            catch (NoSuchElementException e) {
```

Comment:
Can you explain where this is thrown?  I'm just wondering if it's because of the iteration on line 496.

Reply:
So instead of catching an exception, can you instead check if `hasNext()` and throw?